src/charm.py

```diff
@@ -186,22 +191,45 @@ def _on_install(self, _) -> None:
         except (apt.InvalidSourceError, ValueError, apt.GPGKeyError, URLError):
             self.unit.status = BlockedStatus("couldn't install MongoDB")
 
+        # Construct the mongod startup commandline args for systemd
+        machine_ip = self._unit_ip(self.unit)
+        mongod_start_args = " ".join(
+            [
+                "ExecStart=/usr/bin/mongod",
+                # bind to localhost and external interfaces
+                "--bind_ip",
+                f"localhost,{machine_ip}",
+                # part of replicaset
+                "--replSet",
+                "rs0",
+                "\n",
+            ]
+        )
+
+        # mongod.service file passes commandline args to mongod service
+        shutil.copy("src/data/mongod.service", "/etc/systemd/system/mongod.service")
+
+        with open("/etc/systemd/system/mongod.service", "r") as service_template:
+            mongodb_service = service_template.readlines()
+
+        mongodb_service[MONGO_EXEC_LINE] = mongod_start_args
+
+        with open("/etc/systemd/system/mongod.service", "w") as service_file:
+            service_file.writelines(mongodb_service)
```

Comment:
First Comment: yes agreed, I like this, this sounds much better

Second Comment: So I need to do more research on `systemd` and where/when its service files get created. When the install hook executes there is _no_ `mongod.service` file on the machine so one isn't getting overwritten. (But I guess it prevents one from getting created so "in a way" it is overwriting the default file.)

A more ideal scenario would be to grab the default `mongod.service` file and then overwrite that one line. Let me see how I can grab that default file.

Reply:
The default file is written to `/lib/systemd/system/mongod.service`.

Reply:
@marceloneppel Interesting, thanks for sharing! It seems that "Drop-in files in `/etc` take precedence" over those in `/lib`. 

So this solution should work to create a new file while preserving the default file in `/lib/systemd/system/mongod.service`. :D

Reply:
(IMO I think it might be best to read from /lib/systemd/system/mongod.service, modify it, then save it the modified version in /etc/systemd/system/mongod.service)

Yes 👍🏽